dask/blockwise.py

```diff
@@ -64,24 +68,33 @@ def __repr__(self) -> str:
 class BlockwiseDepDict(BlockwiseDep):
     """Base class for dictionary-based Blockwise-IO arguments"""
 
-    def __init__(self, mapping: dict, numblocks: Optional[Tuple[int, ...]] = None):
+    def __init__(
+        self,
+        mapping: dict,
+        numblocks: Optional[Tuple[int, ...]] = None,
+        produces_tasks: Optional[bool] = False,
+    ):
         self.mapping = mapping
+        self.produces_tasks = produces_tasks
 
         # By default, assume 1D shape
         self.numblocks = numblocks or (len(mapping),)
 
     def __getitem__(self, idx: Tuple[int, ...]) -> Any:
         return self.mapping[idx]
 
-    def __dask_distributed_pack__(self, output_blocks):
+    def __dask_distributed_pack__(
+        self, required_indices: Optional[List[Tuple[int, ...]]] = None
+    ):
         from distributed.protocol import to_serialize
 
         return {
             "mapping": {
                 k: to_serialize(self.mapping[k])
-                for k in output_blocks or self.mapping.keys()
+                for k in required_indices or self.mapping.keys()
             },
```

Comment:
nit: when `required_indices` is falsey, could we just use `self.mapping` instead of copying to an intermediate dict?

Also, we might want explicitly check for `required_indices is None`, because if it's just an empty sequence, then that means we don't need to serialize any data.

Reply:
>nit: when required_indices is falsey, could we just use self.mapping instead of copying to an intermediate dict?

I could be wrong in this particular case, but I have run into problems in the past where I was pickling dictionary values in `__dask_distributed_pack__`, and then the same values became "double pickled" when fusing the same layer in a seperate compute/persist operation.  I guess it may be fine when the data is marked as `Serialize`.

> Also, we might want explicitly check for required_indices is None, because if it's just an empty sequence, then that means we don't need to serialize any data.

Yeah - Good call.